### ISSUE TITLE
Problems with visualization of a specific TCGA SM series

### ISSUE THREAD

**Issue by fedorov** [herrmannlab/slim#97]
Something is wrong with this series, as identified by @bcli4d: https://dev-viewer.canceridc.dev/slim/studies/2.25.68803095896966276583382138924964839274/series/1.3.6.1.4.1.5962.99.1.1163866303.1057408148.1637546438847.2.0

Demo https://www.screencast.com/t/AcD9V8DJ7

@dclunie could this be an issue with the data and not the viewer? It seems like high-res layer is either wrong altogether, or is in wrong coordinate space.

**Comment by dclunie** [comment]
It is a problem with the data due to conversion - I can reproduce the same problem displaying the DICOM images in QuPath, whereas the source SVS image looks fine.

The conversion logs shows that there was a problem extracting the JPEG tiles from the SVS and updating them with the factored out quantization and compression tables.

I will investigate further and report back, as well as check if any other images are affected.

**Comment by hackermd** [comment]
Thanks @dclunie for looking into it

**Comment by dclunie** [comment]
It turns out that Aperio creates some SVS files with no compressed JPEG data in a tile at the end of rows, i.e. the TileByteCount is zero. If this is propagated into the DICOM file as a zero length fragment, the reader won't count it as a frame, since there is not a 1:1 correspondence between fragments and frames.

I have updated the converter to create a valid empty compressed tile when there is a zero length TileByteCount in the input, and that seems to work.

I will check for other instances of this in all collections, and figure out how many of these need to be fixed and a strategy for doing so.

**Comment by bcli4d** [comment]
@fedorov you are correct. Replacing some instances would be really difficult at this point. The changes ripple through all parts of the version. I recommend that we should pick up any corrections in the next version.
